### PR TITLE
[fix hooks] libarchive

### PR DESCRIPTION
**Pull request by danimtb**
Specify library name and version:  **libarchive/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by ericLemanissier — DISMISSED** [review]

**Review by Croydon — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by danimtb** [comment]
🤔 I tried this configuration locally and it worked for me... any clue?

**Comment by danimtb** [comment]
This is the error https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4061?view=msvc-160 but I don't why it fails in the CI but not in my machine. Maybe it is due to the version of the compiler?

**Comment by uilianries** [comment]
I'm on Windows right now, lemme try to reproduce it.

**Comment by uilianries** [comment]
My version is a bit older, but I can reproduce it:

```
libarchive/3.4.0: Folder: C:\Users\uilia\.conan\data\libarchive\3.4.0\_\_\export
libarchive/3.4.0: Exported revision: e1dbeefba98103fa6122b58204de7f8e
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=Visual Studio
compiler.runtime=MTd
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

...

libarchive/3.4.0: Calling build()
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19041.
-- The C compiler identification is MSVC 19.27.29111.0
-- The CXX compiler identification is MSVC 19.27.29111.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done

...

C:\Users\uilia\.conan\data\libarchive\3.4.0\_\_\build\d78f64a6ae7b5e68fbc1e982868cb58c2031c65d\source_subfolder\libarchive\archive_read_support_format_warc.c(344,2): error C4061: enumerator 'WT_NONE' in switch of enum 'warc_type_t'
is not explicitly handled by a case label [C:\Users\uilia\.conan\data\libarchive\3.4.0\_\_\build\d78f64a6ae7b5e68fbc1e982868cb58c2031c65d\build_subfolder\source_subfolder\libarchive\archive_static.vcxproj]
C:\Users\uilia\.conan\data\libarchive\3.4.0\_\_\build\d78f64a6ae7b5e68fbc1e982868cb58c2031c65d\source_subfolder\libarchive\archive_read_support_format_warc.c(75): message : see declaration of 'warc_type_t' [C:\Users\uilia\.conan\dat a\libarchive\3.4.0\_\_\build\d78f64a6ae7b5e68fbc1e982868cb58c2031c65d\build_subfolder\source_subfolder\libarchive\archive_static.vcxproj]
```

**Comment by uilianries** [comment]
@danimtb It's a libarchive bug: https://github.com/libarchive/libarchive/pull/1395

libarchive 3.4.0 doesn't support that profile, but 3.5.0 supports according that fix.

**Comment by danimtb** [comment]
thanks for the look into! Indeed it seems a bug from the library, but why it does not raise in my machine?

This is my setup reported by CMake:

```
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19042.
-- The C compiler identification is MSVC 19.25.28614.0
-- The CXX compiler identification is MSVC 19.25.28614.0
```

**Comment by uilianries** [comment]
According to [this discussion](https://discourse.slicer.org/t/libarchive-build-errors/13310/4), it's a limitation with v142 toolset

File: recipes/libarchive/all/conanfile.py [hunk 1]
```diff
@@ -58,6 +58,14 @@ def source(self):
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
+    def validate(self):
```
Comment (uilianries) [inline recipes/libarchive/all/conanfile.py:61]: I can't remember about `validate` is it a new feature?

Comment (danimtb) [reply recipes/libarchive/all/conanfile.py:61]: yes, you can check it here https://blog.conan.io/2020/12/14/New-conan-release-1-32.html

Comment (uilianries) [reply recipes/libarchive/all/conanfile.py:61]: I would add `required_conan_version = ">=1.32.0"`

**Comment by uilianries** [comment]
Okay, our CI is not running Conan 1.32.0. It's blocked.

**Comment by mathbunnyru** [comment]
@danimtb should work well now (https://github.com/conan-io/conan-center-index/blob/master/docs/changelog.md#29-december-2020---1718-cet)
Could you please retrigger the CI?

File: recipes/libarchive/all/conanfile.py (hunk 1 above)
Comment (mathbunnyru) [reply recipes/libarchive/all/conanfile.py:61]: @uilianries may I ask you to add this required_conan_version to the top?

**Review by mathbunnyru — CHANGES_REQUESTED** [review]

**Review by mathbunnyru — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by SpaceIm** [comment]
@danimtb Could you take advantage of this PR to bump dependencies please?

```python
    def requirements(self):
        self.requires("zlib/1.2.11")
        if self.options.with_bzip2:
            self.requires("bzip2/1.0.8")
        if self.options.with_openssl:
            self.requires("openssl/1.1.1d") # => 1.1.1i
        if self.options.with_lz4:
            self.requires("lz4/1.9.2") # => 1.9.3
        if self.options.with_zstd:
            self.requires("zstd/1.4.3") # => 1.4.8
        if self.options.with_lzma:
            self.requires("xz_utils/5.2.4") # => 5.2.5
        if self.options.with_libxml2:
            self.requires("libxml2/2.9.9") # => 2.9.10
        if self.options.with_expat:
            self.requires("expat/2.2.7") # => 2.2.10
        if self.options.with_iconv:
            self.requires("libiconv/1.15") # => 1.16
        if self.options.with_pcreposix:
            self.requires("pcre/8.41") # => 8.44
        # TODO: deps not covered yet: cng, nettle, libb2
```

**Comment by mathbunnyru** [comment]
@SpaceIm I would like it to be done it in another review. I do think that CI is not stable for this recipe, so it would be better to merge what already works.

**Comment by danimtb** [comment]
Agree with @mathbunnyru, I prefer to have this merged and stick to the initial goal of the PR

**Review by uilianries — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]
Ok so let's this PR get merged quickly

**Review by SSE4 — APPROVED** [review]

**Comment by dmn-star** [comment]
This is the line SET(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} /we4061") in libarchive CMakeList.txt that treats the harmless C4061 warning as an error. 

You can simple patch this line out to get libarchive compiled on VS 2019 again.

**Pull request merged**